test_crates/trait_associated_type_added/new/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+#![feature(associated_type_defaults)]
+
+mod sealed {
+    pub(crate) trait Sealed {}
+}
+
+pub trait WillGainTypeWithoutDefault {
+    type Bar;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait WillGainTypeWithoutDefaultSealed: sealed::Sealed {
+    type Bar;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait WillGainTypeWithoutDefaultAndSeal: sealed::Sealed {
+    type Bar;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait WillGainTypeWithDefault {
+    type Bar = bool;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait WillGainTypeWithDefaultSealed: sealed::Sealed {
+    type Bar = bool;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait WillGainAnotherTypeWithoutDefault {
+    type One;
+    type Two;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait WillGainAnotherTypeWithoutDefaultSealed: sealed::Sealed {
+    type One;
+    type Two;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait WillGainADocHiddenType {
+    #[doc(hidden)]
+    type Bar;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait TypeWithoutDefaultUnchanged {
+    type Bar;
+
+    fn make_me_non_object_safe() -> Self;
+}
+
+pub trait TypeDocHidden {
+    #[doc(hidden)]
+    type Bar;
+
+    fn make_me_non_object_safe() -> Self;
+}
```

Comment:
Let's make sure all files end in a trailing newline, for maximum compatibility with other tools. All editors have a setting to do this automatically, so I recommend enabling it. For example, here are the [instructions for VSCode](https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files)
```suggestion
}

```